pyquil/api/_qcs_client.py

```diff
@@ -28,7 +29,18 @@ def qcs_client(*, client_configuration: QCSClientConfiguration, request_timeout:
     :param client_configuration: Client configuration.
     :param request_timeout: Time limit for requests, in seconds.
     """
+    _ensure_event_loop()
     with build_sync_client(
         configuration=client_configuration, client_kwargs={"timeout": request_timeout}
     ) as client:  # type: httpx.Client
         yield client
+
+
+def _ensure_event_loop() -> None:
```

Comment:
When `qcs_client()` is called from a thread other than the main thread, it may not have an event loop. This ensures there is one present.